frontend/src/scenes/cohorts/Cohort.tsx

```diff
@@ -1,199 +1,325 @@
 
-    const onDescriptionChange = (description: string): void => {
+    const onAddGroup = (): void => {
         setCohort({
             ...cohort,
-            description,
+            groups: [
+                ...cohort.groups,
+                {
+                    id: Math.random().toString().substr(2, 5),
+                    matchType: PROPERTY_MATCH_TYPE,
+                    properties: [],
+                },
+            ],
         })
     }
-
-    const onTypeChange = (type: string): void => {
-        if (type === COHORT_STATIC) {
-            setCohort({
-                ...cohort,
-                is_static: true,
-            })
-        } else if (type === COHORT_DYNAMIC) {
-            setCohort({
-                ...cohort,
-                is_static: false,
-            })
-        }
-    }
-
-    const staticCSVDraggerProps = {
-        name: 'file',
-        multiple: false,
-        fileList: cohort.csv ? [cohort.csv] : [],
-        beforeUpload(file: UploadFile) {
-            setCohort({ ...cohort, csv: file })
-
-            return false
-        },
-        accept: '.csv',
+    const onRemoveGroup = (index: number): void => {
+        setCohort({ ...cohort, groups: cohort.groups.filter((_, i) => i !== index) })
     }
 
-    const COHORT_TYPE_OPTIONS = [
-        {
-            key: COHORT_STATIC,
-            label: 'Static',
-            description: 'Upload a list of users. Updates manually',
-            icon: <OrderedListOutlined />,
-        },
-        {
-            key: COHORT_DYNAMIC,
-            label: 'Dynamic',
-            description: 'Cohort updates dynamically based on properties',
-            icon: <CalculatorOutlined />,
-        },
-    ]
-
-    const cohortTypeDropdown = (): JSX.Element => (
-        <DropdownSelector
-            options={COHORT_TYPE_OPTIONS}
-            disabled={cohort.id !== 'new'}
-            value={cohort.is_static ? COHORT_STATIC : COHORT_DYNAMIC}
-            onValueChange={onTypeChange}
-        />
-    )
-
     return (
         <div className="cohort">
-            <PageHeader
-                title={cohort.id === 'new' ? 'New cohort' : cohort.name || 'Untitled'}
-                buttons={
-                    <div className="flex-center">
-                        {cohort.id !== 'new' && (
-                            <LemonButton
-                                data-attr="delete-cohort"
-                                status="danger"
-                                type="secondary"
-                                onClick={() => {
-                                    deleteCohort()
-                                }}
-                                style={{ marginRight: 16 }}
-                            >
-                                Delete
-                            </LemonButton>
+            <VerticalForm logic={cohortLogic} props={logicProps} formKey="cohort">
+                {cohortLoading ? (
```

Comment:
Sorry, I was misleading here. Looking at other pages, I think the pattern we use elsewhere is adding the `loading={cohortLoading}` prop to the save button to give the user feedback on saving.

Reply:
That sounds like a better idea 👍